pkg/webhook/certs.go

```diff
@@ -496,3 +505,30 @@ func (r *ReconcileVWH) Reconcile(request reconcile.Request) (reconcile.Result, e
 
 	return reconcile.Result{}, nil
 }
+
+// ensureValidCerts ensure the certs are created and valid.
+func (cr *CertRotator) ensureValidCerts() {
+	defer close(cr.CertsValidated)
+	checkFn := func() (bool, error) {
+		secret := &corev1.Secret{}
+		if err := cr.client.Get(context.Background(), cr.SecretKey, secret); err != nil {
+			crLog.Error(err, "while retrieving secret")
+			return false, nil
+		}
+		if secret.Data == nil {
+			return false, nil
+		}
+		if cr.validCACert(secret.Data[caCertName], secret.Data[caKeyName]) && cr.validServerCert(secret.Data[caCertName], secret.Data[certName], secret.Data[keyName]) {
+			return true, nil
+		}
+		return false, nil
+	}
+	if err := wait.ExponentialBackoff(wait.Backoff{
+		Duration: 1 * time.Second,
+		Factor:   2,
+		Jitter:   1,
+		Steps:    10,
+	}, checkFn); err != nil {
+		crLog.Error(err, "max retries for checking certs validity")
```

Comment:
This should be a fatal error that causes the `Start()` function to exit with an error so that the master does not hang indefinitely.

Reply:
Good catch, yes. I added a channel `certsNotMounted` from this goroutine to `Start()` and closes the channel here when max retries is reached.